google/cloud/firestore_v1/base_document.py

```diff
@@ -13,17 +13,33 @@
 # limitations under the License.
 
 """Classes for representing documents for the Google Cloud Firestore API."""
+from __future__ import annotations
 
 import copy
-from typing import Any, Dict, Iterable, NoReturn, Optional, Tuple, Union
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Iterable,
+    NoReturn,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+)
 
 from google.api_core import retry as retries
 
 from google.cloud.firestore_v1 import _helpers
 from google.cloud.firestore_v1 import field_path as field_path_module
+from google.cloud.firestore_v1.types import common
 
 # Types needed only for Type Hints
-from google.cloud.firestore_v1.types import Document, common, firestore, write
+if TYPE_CHECKING:  # pragma: NO COVER
+    from google.cloud.firestore_v1.types import Document, firestore, write
+
+
+T = TypeVar("T")
```

Comment:
is this needed?

Reply:
Thanks for catching this. I think I added generic type here at first, but removed because sometimes it's not accepted by python 3.7.

Reply:
removed